x/collection/keeper/grpc_query.go

```diff
@@ -509,6 +509,33 @@ func (s queryServer) Root(c context.Context, req *collection.QueryRootRequest) (
 	return &collection.QueryRootResponse{Root: *token}, nil
 }
 
+func (s queryServer) HasParent(c context.Context, req *collection.QueryHasParentRequest) (*collection.QueryHasParentResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+
+	if err := collection.ValidateContractID(req.ContractId); err != nil {
+		return nil, err
+	}
+
+	if err := collection.ValidateNFTID(req.TokenId); err != nil {
+		return nil, err
+	}
+
+	ctx := sdk.UnwrapSDKContext(c)
+
+	if err := s.validateExistenceOfCollectionGRPC(ctx, req.ContractId); err != nil {
+		return nil, err
+	}
+
+	if err := s.keeper.hasNFT(ctx, req.ContractId, req.TokenId); err != nil {
+		return nil, status.Error(codes.NotFound, err.Error())
+	}
+
+	_, err := s.keeper.GetParent(ctx, req.ContractId, req.TokenId)
```

Comment:
I think it's better to extract another function because it is duplicated. How about this?

Reply:
OK. I will try.